src/hotspot/share/cds/archiveUtils.cpp

```diff
@@ -369,3 +370,34 @@ void ArchiveUtils::log_to_classlist(BootstrapInfo* bootstrap_specifier, TRAPS) {
     }
   }
 }
+
+// Used in logging: "boot", "boot2", "plat", "app" and "unreg";
+const char* ArchiveUtils::class_category(Klass* k) {
+  if (ArchiveBuilder::is_active() && ArchiveBuilder::current()->is_in_buffer_space(k)) {
+    k = ArchiveBuilder::current()->get_source_addr(k);
+  }
+
+  if (k->is_array_klass()) {
+    return "array";
+  } else {
+    oop loader = k->class_loader();
+    if (loader == nullptr) {
+      if (k->module() != nullptr &&
+          k->module()->name() != nullptr &&
+          k->module()->name()->equals("java.base")) {
+        return "boot"; // boot classes in java.base
+      } else {
+        return "boot2"; // boot classes outside of java.base
```

Comment:
Suggestion: boot -> boot-base, boot2 -> boot-nonbase ?

Reply:
I prefer boot/boot2 to make the output easier to read. Anyone debugging this output will need to read the code to understand what "boot2" or "boot-nonbase" is. A few extra characters here will not help.

Reply:
Sorry but '2' conveys zero information whereas 'nonbase' tells you they are not in the base module.

Reply:
It's the second group of boot loaders classes.